src/win32/posix_w32.c

```diff
@@ -402,8 +402,12 @@ int p_symlink(const char *target, const char *path)
 		git__utf8_to_16(target_w, MAX_PATH, target) < 0)
 		return -1;
 
-	if (!CreateSymbolicLinkW(path_w, target_w,
-	    SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE))
+	DWORD dwFlags = SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE;
```

Comment:
We need to declare variables at the beginning of its scope due to C89-compatibility.

Reply:
Hmm
It seems like all of your CI machines compile it fine! (Which is reasonable because all modern compilers support even C++17)
But I'll make the change if that's a requirement.